index.js

```diff
@@ -0,0 +1,161 @@
+const from = require('from2')
+const { EventEmitter } = require('events')
+
+const { Api, JsonRpc } = require('eosjs')
+const { JsSignatureProvider } = require('eosjs/dist/eosjs-jssig') // development only
+const fetch = require('node-fetch') // node only; not needed in browsers
+const { TextEncoder, TextDecoder } = require('util') // node only; native TextEncoder/Decoder
+
+module.exports = configure
+
+function configure (opts) {
+  if (!opts) opts = {}
+  const chainId = opts.chainId || '5fff1dae8dc8e2fc4d5b23b2c7665c97f9e9d8edf2b6485a86ba311c25639191'
+  const rpc = new JsonRpc(opts.rpc || 'https://api-kylin.eoslaomao.com', { fetch })
+
+  let api = null
+  const account = opts.account
+  if (!account) throw new Error('opts.account must be set')
+
+  if (opts.privateKey) {
+    const signatureProvider = new JsSignatureProvider([ opts.privateKey ])
+
+    api = new Api({
+      rpc,
+      signatureProvider,
+      textDecoder: new TextDecoder(),
+      textEncoder: new TextEncoder(),
+      chainId
+    })
+  }
+
+  return {
+    pay,
+    subscription,
+    createTransactionStream
+  }
+
+  function pay (destination, amount, memo, cb) {
+    if (!api) throw new Error('opts.privateKey must be provided in the constructor')
+
+    api.transact({
+      actions: [{
+        account: 'eosio.token',
+        name: 'transfer',
+        authorization: [{
+          actor: account,
+          permission: 'owner'
+        }],
+        data: {
+          from: account,
+          to: destination,
+          quantity: amount,
+          memo: memo
+        }
+      }]
+    }, {
+      blocksBehind: 3,
+      expireSeconds: 30
+    }).then(() => process.nextTick(cb, null)).catch((err) => process.nextTick(cb, err))
+  }
+
+  function subscription (filter, rate) {
+    const match = rate.trim().match(/^(\d(?:\.\d+)?)\s*EOS\s*\/\s*s$/i)
+    if (!match) throw new Error('rate should have the form "n....nn EOS/s"')
+
+    const perSecond = Number(match[1])
+    const sub = new EventEmitter()
+
+    const stream = createTransactionStream()
+    const activePayments = []
+
+    stream.on('data', function (data) {
+      if (data.act.data.memo !== filter) return
+      if (data.act.data.to !== account) return
```

Comment:
is the tx already irreversibe?

take a look here for verifying incoming txs:

```
actions[0].action_trace.act.account == "eosio.token" &&
    actions[0].action_trace.act.name == "transfer" &&
    actions[0].action_trace.act.data.quantity == "X.0000 SYS" &&
    actions[0].action_trace.to == "tokenxchange" && 
    actions[0].action_trace.memo == "KEY TO IDENTIFY INTERNAL ACCOUNT" && 
    actions[0].action_trace.receipt.receiver == "tokenxchange"  &&
    actions[0].block_num < last_irreversible_block
```

(from https://developers.eos.io/eosio-cpp/docs/exchange-deposit-withdraw)

Reply:
ah nice, didn't know :)

Reply:
fixed in latest